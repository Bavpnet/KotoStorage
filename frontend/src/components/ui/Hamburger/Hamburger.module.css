@value mobile-max-width from './../../../globals.css';

.hamburger {
  display: none;
  cursor: pointer;
  background-color: transparent;
  border: none;
}

.bar {
  display: block;
  width: 25px;
  height: 3px;
  margin: 5px auto;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  background-color: #101010;
}

@media only screen and (max-width: mobile-max-width) {
  .hamburger {
    display: block;
    cursor: pointer;
  }

  .hamburger.active .bar:nth-child(2) {
    opacity: 0;
  }

  .hamburger.active .bar:nth-child(1) {
    transform: translateY(8px) rotate(45deg);
  }

  .hamburger.active .bar:nth-child(3) {
    transform: translateY(-8px) rotate(-45deg);
  }
}
