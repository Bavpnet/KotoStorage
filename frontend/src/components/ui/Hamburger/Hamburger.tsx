import classes from './Hamburger.module.css'

type Props = {
  onClick: () => void
  isMenuOpen: boolean
}

const Hamburger: React.FC<Props> = ({ onClick, isMenuOpen }) => {
  return (
    <div
      className={`${classes.hamburger} ${isMenuOpen ? classes.active : ''}`}
      onClick={onClick}
    >
      <span className={classes.bar} />
      <span className={classes.bar} />
      <span className={classes.bar} />
    </div>
  )
}

export default Hamburger
