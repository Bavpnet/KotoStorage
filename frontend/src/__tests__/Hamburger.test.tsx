import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import { Hamburger } from '../components/ui/Hamburger'

describe('Hamburger', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Hamburger isMenuOpen={true} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('change on click', () => {
    const { getByRole } = render(<Hamburger isMenuOpen={true} />)
    const hamburger = getByRole('button')

    // simulate click
    hamburger.click()

    expect(hamburger).toHaveClass('active')
  })
})
