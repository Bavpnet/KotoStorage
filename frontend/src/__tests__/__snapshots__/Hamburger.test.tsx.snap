// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger renders correctly 1`] = `
<button
  className="hamburger active"
>
  <span
    className="bar"
  />
  <span
    className="bar"
  />
  <span
    className="bar"
  />
</button>
`;
